lib/iris/fileformats/pp.py

```diff
@@ -2404,6 +2416,11 @@ def save_pairs_from_cube(cube, field_coords=None, target=None):
 
     # Save each named or latlon slice2D in the cube
     for slice2D in cube.slices(field_coords):
+        # Attach an extra cube "fill_value" property, allowing the save rules
+        # to deduce MDI more easily without realising the data.
+        # NOTE: it is done this way because this property may exist in future.
+        slice2D.fill_value = _data_fill_value(slice2D)
```

Comment:
This is a purely temporary measure, based on the idea that the forthcoming dask change will definitely be introducing the generic "cube.fill_value" property.
We will remove this hack when the dask branch is merged back.
Obviously, the method actually used here is :nauseated_face: but it is only here so you guys can check that it fixes the current memory burn problem.